app/models/thesis.rb

```diff
@@ -146,6 +146,15 @@ class Thesis < ApplicationRecord
 
   enum proquest_exported: ['Not exported', 'Full harvest', 'Partial harvest']
 
+  # Looks up the thesis' accession number based on its degree period.
+  def accession_number
+    degree_period = DegreePeriod.find_by(grad_year: graduation_year, grad_month: graduation_month)
```

Comment:
I intended for this to be a much more thorough review, but that's not looking likely at this point this afternoon. I did want to at least ask this question, with the caveat that I may have forgotten too much of Rails for it to be meaningful:

It makes me a bit nervous to see basically the entire `look_up_degree_period` helper repeated in the Thesis model - is it possible for this to use the helper, rather than replicate it?

Reply:
You can't directly call a Helper from a Model. This is where some people abstract this out to a Concern.

Reply:
Or maybe in this case `look_up_degree_period` could actually be in the model itself. I haven't made it deep enough into the review to see how it is used in the UI.